packages/modules/viewer-react/src/types.ts

```diff
@@ -102,6 +104,10 @@ export interface LoaderProps {
   viewCreatorOptions?: ViewerViewCreator3dOptions;
   /** Component to show when loading iModel key */
   loadingComponent?: React.ReactNode;
+  /** unified selection storage to synchronize with. Should be passed together with `getSchemaContext` */
```

Comment:
Yes and no... 

The `useUnifiedSelectionSync` makes sure that `IModelConnection` selection changes are synchronized with the new unified selection storage; the hook is used by `IModelLoader`, which means we need those two props on `LoaderProps`.

However, we also need `Presentation` library to use the exact same storage to make sure that components using `SelectionManager` from `@itwin/presentation-frontend` get notified about changes in the storage and can change its content. This is done by passing the storage to `Presentation.initialize` call, which is done by the `BaseInitializer.initialize`, which takes `ViewerInitializerParams`. So we need at least `selectionStorage` prop on `ViewerInitializerParams`.

Here's what I suggest:
- Revert the current change and move the two props to a new interface `UnifiedSelectionProps`.
- In `ViewerInitializerParams` keep only `selectionStorage?: UnifiedSelectionProps["selectionStorage"]`. I think this could be avoided, because `useBaseViewerInitializer` receives selection storage and could inject it into `PresentationProps` that are passed to `BaseInitializer.initialize`. However, I have a hard time understanding what's happening in `useBaseViewerInitializer` to make any changes there confidently.
- Extend `ModelLoaderProps` with `& UnifiedSelectionProps`.
- Change `ViewerCommonProps` to `Omit<ViewerInitializerParams, "selectionStorage"> & LoaderProps & UnifiedSelectionProps`.

Reply:
I think your suggestions sound good, I'll review the useBaseViewerInitializer changes you're proposing, like you said not sure if needed

Reply:
I think I managed to supply the required props without introducing anything new to `ViewerInitializerParams`. Please have a look.

Reply:
lgtm, thx